storm-client/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -52,16 +64,19 @@ public JcMetrics jcMetrics(String name, String topologyId, String componentId, I
 
     public Meter meter(String name, WorkerTopologyContext context, String componentId, Integer taskId, String streamId) {
         String metricName = metricName(name, context.getStormId(), componentId, streamId, taskId, context.getThisWorkerPort());
+        LOG.warn("Adding meter: " + metricName);
         return registry.meter(metricName);
     }
 
     public Counter counter(String name, WorkerTopologyContext context, String componentId, Integer taskId, String streamId) {
         String metricName = metricName(name, context.getStormId(), componentId, streamId, taskId, context.getThisWorkerPort());
+        LOG.warn("Adding counter: " + metricName);
         return registry.counter(metricName);
```

Comment:
Since the registry is shared among tasks inside the same worker, all the metrics should have different name otherwise the same metric object will be returned: https://github.com/dropwizard/metrics/blob/4.1-development/metrics-core/src/main/java/com/codahale/metrics/MetricRegistry.java#L172-L181

So if `registery.counter("a-counter", xxxx)` is called more than once, starting from the second time, it returns the same metric object. 

Which means all tasks will have the same metric objects in the worker. For example, all tasks have `acked`, `failed`, `emitted`, `transferred` metrics
https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/metrics2/TaskMetrics.java#L21-L24

With this code change, it will return the same value for all the tasks inside the same worker.